### ISSUE TITLE
[1.20] Wrong images are used in static pods

### ISSUE THREAD

**Issue by rancher-max** [rancher/rke2#1443]
**Environmental Info:**
RKE2 Version: 
v1.20.9-rc4+rke2r1

Node(s) CPU architecture, OS, and Version: 
Centos8

Cluster Configuration:
3 server 1 agent

**Describe the bug:**

The images used in pods are not all correct. See list from a fresh install (this is the same on upgraded setup). Notice specifically the **rke2r2** for `kube-apiserver`, `kube-controller-manager`, `kube-proxy`, and `kube-scheduler`
```
$ kubectl get pods --all-namespaces -o=jsonpath='{range .items[*]}{"\n"}{.metadata.name}{":\t"}{range .spec.containers[*]}{.image}{", "}{end}{end}' |\
> sort

etcd-ip-172-31-12-164.us-east-2.compute.internal:	index.docker.io/rancher/hardened-etcd:v3.4.13-k3s1-build20210223, 
etcd-ip-172-31-3-237.us-east-2.compute.internal:	index.docker.io/rancher/hardened-etcd:v3.4.13-k3s1-build20210223, 
etcd-ip-172-31-8-83.us-east-2.compute.internal:	index.docker.io/rancher/hardened-etcd:v3.4.13-k3s1-build20210223, 
helm-install-rke2-canal-5kkgk:	rancher/klipper-helm:v0.5.0-build20210505, 
helm-install-rke2-coredns-knhqd:	rancher/klipper-helm:v0.5.0-build20210505, 
helm-install-rke2-ingress-nginx-jpxjd:	rancher/klipper-helm:v0.5.0-build20210505, 
helm-install-rke2-metrics-server-2mrvh:	rancher/klipper-helm:v0.5.0-build20210505, 
kube-apiserver-ip-172-31-12-164.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
kube-apiserver-ip-172-31-3-237.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
kube-apiserver-ip-172-31-8-83.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
kube-controller-manager-ip-172-31-12-164.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
kube-controller-manager-ip-172-31-3-237.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
kube-controller-manager-ip-172-31-8-83.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
kube-proxy-ip-172-31-12-111.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
kube-proxy-ip-172-31-12-164.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
kube-proxy-ip-172-31-3-237.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
kube-proxy-ip-172-31-8-83.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
kube-scheduler-ip-172-31-12-164.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
kube-scheduler-ip-172-31-3-237.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
kube-scheduler-ip-172-31-8-83.us-east-2.compute.internal:	index.docker.io/rancher/hardened-kubernetes:v1.20.9-rke2r2-build20210716, 
rke2-canal-hsnqt:	rancher/hardened-calico:v3.13.3-build20210223, rancher/hardened-flannel:v0.13.0-rancher1-build20210223, 
rke2-canal-klv79:	rancher/hardened-calico:v3.13.3-build20210223, rancher/hardened-flannel:v0.13.0-rancher1-build20210223, 
rke2-canal-m78qk:	rancher/hardened-calico:v3.13.3-build20210223, rancher/hardened-flannel:v0.13.0-rancher1-build20210223, 
rke2-canal-mmvnn:	rancher/hardened-calico:v3.13.3-build20210223, rancher/hardened-flannel:v0.13.0-rancher1-build20210223, 
rke2-coredns-rke2-coredns-675b5596bc-q5hxt:	rancher/hardened-coredns:v1.6.9-build20210223, 
rke2-ingress-nginx-controller-54946dd48f-l5pts:	rancher/nginx-ingress-controller:nginx-0.30.0-rancher1, 
rke2-ingress-nginx-default-backend-5795954f8-954ml:	rancher/nginx-ingress-controller-defaultbackend:1.5-rancher1, 
rke2-metrics-server-559dc47bd8-sw9th:	rancher/hardened-k8s-metrics-server:v0.3.6-build20210223, 
```

I noticed this when upgrading the cluster from v1.20.8+rke2r1 to v1.20.9-rc4+rke2r1 via rancher, it appears the cluster version becomes v1.20.9+rke2r2, even though nodes show v1.20.9+rke2r1. See screenshots:
![image](https://user-images.githubusercontent.com/55997940/126589829-c5786911-4c41-43cb-ad2d-d019b3a9e971.png)

![image](https://user-images.githubusercontent.com/55997940/126589841-0c67933f-9865-4888-8ae1-6be1e5a696b5.png)

It looks like something is wrong with the image tags:
![image](https://user-images.githubusercontent.com/55997940/126589906-7f8b879b-a722-46e5-ad1d-c298d0fa4133.png)

**Steps To Reproduce:**

- Installed RKE2 v1.20.9-rc4+rke2r1

**Expected behavior:**

The images should not be r2 and rancher should show the correct version when doing an upgrade. This is based on kube-proxy image version I believe.

**Additional context / logs:**

**Comment by rancher-max** [comment]
This is also causing rancher to try to upgrade the nodes upon importing a fresh install into rancher or running rancher inside the cluster (local cluster)

**Comment by brandond** [comment]
This is my mistake, I missed a variable when updating the image lists for 1.20. will have a fix up shortly.

**Comment by rancher-max** [comment]
Validated this has been fixed in v1.20.9-rc5+rke2r1
